src/yunohost/hook.py

```diff
@@ -427,13 +451,36 @@ def hook_exec(path, args=None, raise_on_error=False, no_trace=False,
     return returncode, returncontent
 
 
+def _hook_exec_python(path, args, env, loggers):
+
+    dir_ = os.path.dirname(path)
+    name = os.path.splitext(os.path.basename(path))[0]
+
+    if not dir_ in sys.path:
+        sys.path = [dir_] + sys.path
+    module = import_module(name)
+
+    # TODO : We might want to check here that it's a tuple
+    # containing an int + a dict ?
+    ret = module.main(args, env, loggers)
```

Comment:
Shouldn't we spawn a new python process that launch the python instead of calling it in the current python process? That would be better security and bug wise I think (but idk how hard would it be to do that)

Reply:
Hm I kinda chose to do this because that's supposedly less ressource expensive, considering that we're likely to import code from yunohost's in each of those (e.g. 10~15ish hooks in the context of diagnosis), each likely to do stuff like importing `requests` or other libs that take hell of time to load on RPi for example